File: agent/sim.py
#!/usr/bin/env python3
"""Python simulation for Claude Operator (minimal)
Run: python -m agent.sim
"""
import os
import random
import json
from datetime import datetime

SIM = os.getenv('SIMULATION_MODE', 'true').lower() == 'true'
VOTE = os.getenv('VOTE_ACCOUNT', 'GEgnNd6AyNshufepHS2HVXhDSRKwhjqgAYvWSnKCyQWf')
COMM_BPS = int(os.getenv('COMMISSION_BPS', '70'))


def sample_metrics():
    return {
        'slotLag': random.randint(0, 10),
        'delinquent': random.random() < 0.02,
        'cpuPct': random.randint(1, 95),
        'memPct': random.randint(40, 98),
        'diskPct': random.randint(10, 95),
        'rpcHealthy': random.random() > 0.05,
    }


def analyze(m):
    issues = []
    if m['delinquent']:
        issues.append('DELINQUENT')
    if m['slotLag'] > 4:
        issues.append('SLOT_LAG')
    if m['cpuPct'] > 85:
        issues.append('HIGH_CPU')
    if m['memPct'] > 90:
        issues.append('HIGH_MEM')
    if m['diskPct'] > 90:
        issues.append('DISK_PRESSURE')
    if not m['rpcHealthy']:
        issues.append('RPC_FAIL')
    return issues


def suggest_actions(issues):
    actions = []
    for it in issues:
        if it in ('DELINQUENT', 'SLOT_LAG'):
            actions.append('restart-validator-service')
        if it in ('HIGH_CPU', 'HIGH_MEM'):
            actions.append('investigate-processes')
        if it == 'DISK_PRESSURE':
            actions.append('rotate-logs-and-clean')
        if it == 'RPC_FAIL':
            actions.append('restart-rpc-service')
    if not issues:
        actions.append('no-op')
    return list(dict.fromkeys(actions))


def main():
    metrics = sample_metrics()
    issues = analyze(metrics)
    actions = suggest_actions(issues)

    report = {
        'timestamp': datetime.utcnow().isoformat() + 'Z',
        'vote': VOTE,
        'metrics': metrics,
        'issues': issues,
        'proposed_actions': actions,
        'simulation': SIM,
    }

    print('CLAUDATOR OPERATOR SIMULATION')
    print(json.dumps(report, indent=2))
    print('\nProposed actions:')
    for a in actions:
        if SIM:
            print('-', '[SIMULATION] would run:', a)
        else:
            print('-', '[EXECUTE] running:', a)

    print('\nCommission rate:', COMM_BPS / 100, '% (bps=', COMM_BPS, ')')


if __name__ == '__main__':
    main()

File: agent/sim.js
#!/usr/bin/env node
// Simulation mode for Claude Operator (Node.js)
const fs = require('fs');
const path = require('path');
const envPath = path.resolve(process.cwd(), '.env');

require('dotenv').config({ path: envPath });

const SIM = (process.env.SIMULATION_MODE || 'true') === 'true';
const VOTE = process.env.VOTE_ACCOUNT || 'GEgnNd6AyNshufepHS2HVXhDSRKwhjqgAYvWSnKCyQWf';
const COMM_BPS = parseInt(process.env.COMMISSION_BPS || '70', 10);

function sampleMetrics() {
  return {
    slotLag: Math.floor(Math.random() * 10),
    delinquent: Math.random() < 0.02,
    cpuPct: Math.floor(Math.random() * 90),
    memPct: Math.floor(50 + Math.random() * 45),
    diskPct: Math.floor(20 + Math.random() * 70),
    rpcHealthy: Math.random() > 0.05
  };
}

function analyze(m) {
  const issues = [];
  if (m.delinquent) issues.push('DELINQUENT');
  if (m.slotLag > 4) issues.push('SLOT_LAG');
  if (m.cpuPct > 85) issues.push('HIGH_CPU');
  if (m.memPct > 90) issues.push('HIGH_MEM');
  if (m.diskPct > 90) issues.push('DISK_PRESSURE');
  if (!m.rpcHealthy) issues.push('RPC_FAIL');
  return issues;
}

function suggestActions(issues) {
  const actions = [];
  for (const it of issues) {
    if (it === 'DELINQUENT' || it === 'SLOT_LAG') actions.push('restart-validator-service');
    if (it === 'HIGH_CPU' || it === 'HIGH_MEM') actions.push('investigate-processes');
    if (it === 'DISK_PRESSURE') actions.push('rotate-logs-and-clean');
    if (it === 'RPC_FAIL') actions.push('restart-rpc-service');
  }
  if (issues.length === 0) actions.push('no-op');
  return [...new Set(actions)];
}

function postAlertSim(event) {
  console.log('[ALERT-SIM]', JSON.stringify(event, null, 2));
}

async function main() {
  console.log('$Claudator Claude Operator — Simulation mode:', SIM);
  console.log('Vote account:', VOTE, 'Commission bps:', COMM_BPS);

  const metrics = sampleMetrics();
  const issues = analyze(metrics);
  const actions = suggestActions(issues);

  const report = {
    timestamp: new Date().toISOString(),
    vote: VOTE,
    metrics,
    issues,
    proposed_actions: actions,
    simulation: SIM
  };

  postAlertSim(report);

  console.log('\nProposed remediation actions:');
  for (const a of actions) {
    if (SIM) {
      console.log('- [SIMULATION] would run:', a);
    } else {
      console.log('- [EXECUTE] running:', a);
      // Real action code would go here (systemctl, log rotate, etc.)
    }
  }

  console.log('\nCommission tracking sample:');
  console.log(`- Commission rate: ${COMM_BPS / 100}% (bps=${COMM_BPS})`);
  console.log('- Simulated path: /reinvestment/receipts/YYYYMMDD-HHMMSS.json');
}

main().catch(e => { console.error(e); process.exit(1); });
